### ISSUE TITLE
Editing Markdown: autocomplete-plus ignores minimum word length and starts already after first character

### ISSUE THREAD

**Issue by bentolor** [atom/autocomplete-plus#501]
I'm currently editing a `file.md`  and the autocomplete-plus extension get's in my way all the time. I'm on Atom 0.208.0 and have not changed any autocomplete-plus settings.

autocomplete-plus  starts proposals after the first keystroke and imposes a _significant_ delay on typing. The settings dialog says the default minimum word length should be 3.

![ignored-minimum-word](https://cloud.githubusercontent.com/assets/588260/8107139/684595ec-1047-11e5-97be-16ba454d8d51.png)

**Comment by benogle** [comment]
This is really two separate issues. I can reproduce the minWordLength issue. 

> imposes a significant delay on typing

Can you make another issue and answer some questions?

Do you have any other files open? Are they large? Is the delay still significant if you just open a single small file?

Can you take a CPU profile for me? See https://atom.io/docs/latest/hacking-atom-debugging#diagnose-performance-problems-with-the-dev-tools-cpu-profiler

**Comment by bentolor** [comment]
Thank you @benogle for your feedback. Glad you were able to reproduce my minWordLength issue. 

Currently I'm not sure if the delay is rather a side effect of the excessive garbage collection overhead in general as demonstrated in https://github.com/atom/atom/issues/6303#issuecomment-111170501 See profiler screenshots there.

The excessive sluggishness  also occurs with autocomplete-plus disabled. I assume it might be only an amplifying effect of this plugin. 

Only one file opened. Pure Markdown file with ~500 lines / 2500 words. Occurs in smaller files only after longer time of usage. I'm currently evaluating the effect in `--safe` an will report back as soon as I have meaningful, new results. Thank for the link!

**Comment by JormaSnow** [comment]
I am getting autocomplete suggestions from the very first keystroke no matter what I set Minimum Word Length to. I found this issue closed but I don't see that it's been solved since it's currently happening to me.

**Comment by benogle** [comment]
@JormaSnow is this with the default provider? Try with atom in [safe mode](https://atom.io/docs/latest/hacking-atom-debugging#check-if-the-problem-shows-up-in-safe-mode)?

**Comment by JormaSnow** [comment]
Sorry, I wasn't watching this.
Can't say as to it happening in safe mode. I only got errors when I try and open anything '--safe.'
Haven't been using Atom for a few weeks, so didn't bother with trying to work through any errors.

**Comment by JormaSnow** [comment]
I had to open --safe as su. In safe mode I do get snippets after a single keystroke, but the other suggestions seem to be quieted. For instance:

'd' lists "df defer statement"

When not in safe mode 'd' returns:
"df defer statement"
"const dbHost"
"const dbName"
"const dbPassword"
"const dbUser"  //obviously I'm testing a database app :)
"deleteRecord(w http.responseWriter, r *http.Request)
